File: judo-data/src/main/java/com/judopay/api/ApiError.java
package com.judopay.api;

class ApiError {

    private final int code;
    private final String fieldName;
    private final String message;
    private final String detail;

    public ApiError(int code, String fieldName, String message, String detail) {
        this.code = code;
        this.fieldName = fieldName;
        this.message = message;
        this.detail = detail;
    }

    public int getCode() {
        return code;
    }

    public String getFieldName() {
        return fieldName;
    }

    public String getMessage() {
        return message;
    }

    public String getDetail() {
        return detail;
    }
}

File: judokit-android/src/main/java/com/judopay/judokit/android/api/error/ApiError.kt
package com.judopay.judokit.android.api.error

import com.judopay.judokit.android.model.JudoError

data class ApiError(
    val code: Int,
    val category: Int,
    val message: String,
    val details: List<ApiErrorDetail>? = emptyList()
) {
    override fun toString(): String {
        return "ApiError(code=$code, category=$category, message='$message', details=$details)"
    }
}

data class ApiErrorDetail(
    val code: Int,
    val message: String,
    val fieldName: String
) {
    override fun toString(): String {
        return "ApiErrorDetail(code=$code, message='$message', fieldName='$fieldName')"
    }
}

fun ApiError.toJudoError() = JudoError(code, message)
